### ISSUE TITLE
Selecting new value using <select> editor loses table focus

### ISSUE THREAD

**Issue by Zuur** [olifolkerd/tabulator#436]
Hi there, what a great tool!

I'm wanting to build a table which users can move around in using the arrow-keys like you do in Excel.

The most major issue I'm having with keyboard-only navigation is that if I implement a custom select editor for a column then selecting a value loses focus on the table completely. I've been trying to restore focus to the table cell once the selection is made but haven't been able to...  Something like this?

    editor.on("change blur", function(e){
        success(editor.val());
        $(cell).focus(); //obviously not this, but what to do here?
    });

I'd also like to implement my own custom editors for text boxes, etc. so that the user can use the arrow keys to move left/right for example instead of it moving the cursor left/right within the edit box. How would I set the focus to be on the cell to the left / right / above or below?

**Comment by olifolkerd** [comment]
Hey,

If you can post a copy of your custom select editor code i will be happy to post a solution for you.

In terms of navigation there are a number of options, you can use the included ***keybindings*** to bind keyboard events directly to table actions:

```js
$("#example-table").tabulator({
    keybindings:{
        "navLeft" : "37", //left arrow key
        "navRight" : "39", //Right arrow key
    },
});
```

See the [Keybindings Documentation](http://olifolkerd.github.io/tabulator/docs/#keybindings) for more info.

Alternatively you can use the navigation function on the cell component passed into the editor to automatically shift the focus:
```js
cell.nav().left();
cell.nav().right();
```
See the [Cell Component Documentation](http://olifolkerd.github.io/tabulator/docs/#component-cell) for more information.

Let me know if that answers your questions.

Cheers

Oli

**Comment by Zuur** [comment]
That's Great thanks Oli! - the cell.nav().next() and .prev() answered a whole bunch of problems for me.

I've got movableColumns enabled which don't play nicely with next() and prev() at the moment (next/prev don't realize they've moved so they jump around in the wrong order). Can you suggest any way to work around this?

**Comment by olifolkerd** [comment]
Hey,

That is a great point, i hadn't though about that aspect of movable columns before. i will put an update into the 3.2 release at the end of the month to fix that behaviour.

Cheers

Oli

**Comment by Zuur** [comment]
Thanks Oli, I've made some great progress with the item you mentioned above and will look forward to the fix in the 3.2 release!

**Comment by olifolkerd** [comment]
Hey,

Tabulator 3.2 has just been released and includes a fix for the navigation functions after row reorder.

Thanks for the feedback!

Cheers

Oli
